client/src/ExtensionState.ts

```diff
@@ -66,7 +65,7 @@ export class State {
 
 
   // creates the language client and starts the server
-  public static startLanguageServer(fileSystemWatcher: vscode.FileSystemWatcher): Promise<any> {
+  public static async startLanguageServer(context: vscode.ExtensionContext, fileSystemWatcher: vscode.FileSystemWatcher): Promise<void> {
```

Comment:
You're not actually using `await` in this function (see comment at the end of function), no need to make it `async`.